### DISCUSSION TITLE
Any thoughts/plans to generate the mapping code?

### DISCUSSION THREAD

**Discussion by dude-dw** [LuckyPennySoftware/AutoMapper#4156] [NA]
Hello!

Do you have plans to _generate_ the mapping code, which can then be compiled (instead of mapping using runtime reflection)? Perhaps I'm naïve regarding the complexity involved, but it would seem to me that having generated code that can be compiled could improve some issues. For example:

- run time performance
- ability to debug
- references would be found in static code analysis
- ability to migrate away from using AutoMapper on a class-by-class basis by "taking over" the generated code for a class when that class grows in complexity

Thank you!

**Comment by jbogard** [comment]
This already exists:

https://mappinggenerator.net/

I've heard great things about it and it sounds like it's what you're looking for.
